version: "3.9"

volumes:
  prefect:
  postgres:

networks:
  default:
    name: prefect-network

services:
  prefect-database:
    image: postgres:16.0-alpine
    restart: unless-stopped
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - POSTGRES_DB=prefect
    ports:
      - 5432:5432
    volumes:
      - postgres:/var/lib/postgresql/data

  server:
    image: prefecthq/prefect:2.13.1-python3.11
    restart: unless-stopped
    volumes:
      - prefect:/root/.prefect
    entrypoint: ["prefect", "server", "start","--host", "0.0.0.0", "--port", "4200"]
    environment:
      - PREFECT_UI_URL=http://127.0.0.1:4200/api
      - PREFECT_API_URL=http://127.0.0.1:4200/api
      #- PREFECT_UI_URL=http://192.168.56.101:4200/api
      #- PREFECT_API_URL=http://192.168.56.101:4200/api
      - PREFECT_SERVER_API_HOST=0.0.0.0
      - PREFECT_API_DATABASE_CONNECTION_URL=postgresql+asyncpg://postgres:postgres@prefect-database:5432/prefect
    ports:
      - 4200:4200
    depends_on:
      - prefect-database

  cli:
    image: prefecthq/prefect:2.13.1-python3.11
    entrypoint: "bash"
    environment:
      - PREFECT_API_URL=http://server:4200/api

      # for maintanance purposes, the cli should have access to the db
      - PREFECT_API_DATABASE_CONNECTION_URL=postgresql+asyncpg://postgres:postgres@prefect-database:5432/prefect
